flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -761,7 +760,9 @@ private CompiledPlan compilePlanAndWrite(
                             + ". This is a bug, please file an issue.");
         }
 
-        compiledPlan.writeToFile(file, false);
+        compiledPlan.writeToFile(localPath, false);
+        resourceManager.updateFilePath(filePath);
```

Comment:
Correct me if I'm wrong, but I think we should not and probably cannot write files to a remote FileSystem. 

Although users have configured related conf in flink-conf.yaml, such as accessKey and accessSecret, having permission to read files under a bucket does not mean having the same write permission. 

If users feel it is necessary, they can upload files manually. Implementing this through the framework may have many limitations.